server/src/test/java/org/elasticsearch/snapshots/SnapshotInfoTests.java

```diff
@@ -0,0 +1,138 @@
+/*
+ * Licensed to Elasticsearch under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.elasticsearch.snapshots;
+
+import org.elasticsearch.common.io.stream.Writeable;
+import org.elasticsearch.index.shard.ShardId;
+import org.elasticsearch.test.AbstractWireSerializingTestCase;
+import org.elasticsearch.test.ESTestCase;
+
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+public class SnapshotInfoTests extends AbstractWireSerializingTestCase<SnapshotInfo> {
```

Comment:
Not sure adding this test adds much tbh., we're already checking that a simple map works in the integration tests now.
If anything, I'd rather add a test that shows that nested maps/lists actually work with our serialization logic here to prevent future breakage should we ever change anything about the serialization format we use.

Reply:
+1 for writing a test that shows that nested maps work fine

Reply:
Good point, I'll adjust this to use a nested map here.